File: 백준/Silver/1094. 막대기/막대기.py
import sys

input = sys.stdin.readline
x = int(input())
stick = [64]

while sum(stick) != x:
    small = stick.pop()
    half = small // 2
    stick.append(half)
    if sum(stick) < x:
        stick.append(half)
print(len(stick))

File: 백준/Silver/1094. 막대기/막대기.swift
let X = Int(readLine()!)!

var sticks = [64]
var total = 64

while total > X {
    sticks.sort()

    let small = sticks.removeFirst()
    let half = small / 2
    sticks.append(half)
    total = total - small + half
    if total < X {
        sticks.append(half)
        total += half
    }
}
print(sticks.count)